Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,152 @@ This document includes four main sections:
+As an exercise, try querying ``sqlite_master`` for a bogus name ``"abc"``::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master where name='abc'")
+   >>> res.fetchone()
+   >>>
+
+As expected, the query returns an empty result.
+
+Now, add two rows of data supplied as SQL literals
+by executing an ``INSERT`` statement,
+once again by calling :meth:`cur.execute(...) <Cursor.execute>`::
 
-At this point, our database only contains one row::
+   cur.execute("""
+       INSERT INTO Movie VALUES
+           ('Monty Python and the Holy Grail', 1975, 8.2),
+           ('And Now for Something Completely Different', 1971, 7.5)
+   """)
 
-   >>> res = cur.execute('SELECT count(rowid) FROM stocks')
-   >>> print(res.fetchone())
-   (1,)
+The ``INSERT`` statement implicitly opens a transaction,
+which needs to be committed before changes are saved in the database
+(see :ref:`sqlite3-controlling-transactions` for details).
+Call :meth:`con.commit() <Connection.commit>` on the connection object
+to commit the transaction::
 
-The result is a one-item :class:`tuple`:
-one row, with one column.
-Now, let us insert three more rows of data,
-using :meth:`~Cursor.executemany`::
+   con.commit()
+
+Execute a query to verify that the data was inserted correctly.
+Use the now familiar :meth:`con.execute(...) <Cursor.execute>` to
+store the result in ``res``,
+and call :meth:`res.fetchall() <Cursor.fetchall>` to fetch all rows::
+
+   >>> res = cur.execute("SELECT Score FROM Movie")
+   >>> res.fetchall()
+   [(8.2,), (7.5,)]
+
+The result is a :class:`list` with two rows of data,
+each a one-item :class:`!tuple`,
+containing the value from the single column queried.
 
-   >>> data = [
-   ...    ('2006-03-28', 'BUY', 'IBM', 1000, 45.0),
-   ...    ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0),
-   ...    ('2006-04-06', 'SELL', 'IBM', 500, 53.0),
-   ... ]
-   >>> cur.executemany('INSERT INTO stocks VALUES(?, ?, ?, ?, ?)', data)
+Now, insert three more rows by calling
+:meth:`cur.executemany(...) <Cursor.executemany>`::
 
-Notice that we used ``?`` placeholders to bind *data* to the query.
+   data = [
+       ("Monty Python Live at the Hollywood Bowl", 1982, 7.9),
+       ("Monty Python's The Meaning of Life", 1983, 7.5),
+       ("Monty Python's Life of Brian", 1979, 8.0),
+   ]
+   cur.executemany("INSERT INTO Movie VALUES(?, ?, ?)", data)
+   con.commit()  # Remember to commit the transaction after executing INSERT.
+
+Notice that ``?`` placeholders are used to bind ``data`` to the query.
 Always use placeholders instead of :ref:`string formatting <tut-formatting>`
 to bind Python values to SQL statements,
 to avoid `SQL injection attacks`_.
-See the :ref:`placeholders how-to <sqlite3-placeholders>` for more details.
-
-Then, retrieve the data by iterating over the result of a ``SELECT`` statement::
-
-   >>> for row in cur.execute('SELECT * FROM stocks ORDER BY price'):
+See :ref:`sqlite3-placeholders` for more details.
```

Comment:
For consistency with the transaction above, this should be `(see ... for details).`